kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ingest/client/MySQLClient.java

```diff
@@ -305,15 +309,25 @@ private final class MySQLBinlogEventHandler extends ChannelInboundHandlerAdapter
             this.lastBinlogEvent = new AtomicReference<>(lastBinlogEvent);
         }
         
+        @SuppressWarnings("unchecked")
         @Override
         public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
             if (!running) {
                 return;
             }
+            reconnectTimes.set(0);
+            if (msg instanceof List) {
+                List<AbstractBinlogEvent> records = (List<AbstractBinlogEvent>) msg;
+                if (records.isEmpty()) {
+                    log.warn("The records is empty");
+                    return;
+                }
+                lastBinlogEvent.set(records.get(records.size() - 1));
+                blockingEventQueue.put(records);
```

Comment:
Does `records` include whole transaction's records?

Reply:
Yes, if the msg is List, it's include whole transaction's records.

Reply:
When is `MySQLClient.MySQLBinlogEventHandler.channelRead` invoked, looks it's not guaranteed

Reply:
It's guaranteed by previous handler `MySQLBinlogEventPacketDecoder`